application/dash_application/utility/plot_builder.py

```diff
@@ -309,11 +389,72 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
                 'title': {
                     'text': axis_text
                 }
+            },
+            legend={
+                'tracegroupgap': 0
             }
         )
     )
 
 
+def generate_subplot(subplots: List[Subplot]):
```

Comment:
I think this was a source of confusion for me - the terminology plotly uses is that one plot holds multiple subplots, it seems like we've been saying one subplot holds multiple subplots? So 'generate_subplot' to me suggests it generates just 1 of the subplots, not all of them